commit e2898a3d1db00dd0e68eb30cd0b6c707731f62e8
Author: Christian Brauner <christian.brauner@ubuntu.com>
Date:   Sun Jul 2 23:31:18 2017 +0200

    import: keep volatile keys
    
    We shouldn't wipe volatile keys on container import. Otherwise we'll lose
    things like hwaddr and last idmapping.
    
    Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

diff --git a/lxd/api_internal.go b/lxd/api_internal.go
index 2715718..3a520be 100644
--- a/lxd/api_internal.go
+++ b/lxd/api_internal.go
@@ -8,7 +8,6 @@ import (
 	"net/http"
 	"os"
 	"strconv"
-	"strings"
 
 	"github.com/gorilla/mux"
 	"gopkg.in/yaml.v2"
@@ -345,11 +344,6 @@ func internalImport(d *Daemon, r *http.Request) Response {
 	}
 
 	baseImage := backup.Container.Config["volatile.base_image"]
-	for k := range backup.Container.Config {
-		if strings.HasPrefix(k, "volatile") {
-			delete(backup.Container.Config, k)
-		}
-	}
 
 	arch, err := osarch.ArchitectureId(backup.Container.Architecture)
 	if err != nil {
@@ -430,11 +424,6 @@ func internalImport(d *Daemon, r *http.Request) Response {
 		}
 
 		baseImage := snap.Config["volatile.base_image"]
-		for k := range snap.Config {
-			if strings.HasPrefix(k, "volatile") {
-				delete(snap.Config, k)
-			}
-		}
 
 		arch, err := osarch.ArchitectureId(snap.Architecture)
 		if err != nil {
